modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/node/NodeExecutorRegistry.java

```diff
@@ -14,62 +14,42 @@
 
 package com.liferay.portal.workflow.kaleo.runtime.internal.node;
 
-import com.liferay.portal.workflow.kaleo.definition.NodeTypeDependentObjectRegistry;
-import com.liferay.portal.workflow.kaleo.definition.exception.KaleoDefinitionValidationException;
+import com.liferay.osgi.service.tracker.collections.map.ServiceReferenceMapperFactory;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.workflow.kaleo.definition.NodeType;
 import com.liferay.portal.workflow.kaleo.runtime.node.NodeExecutor;
 
-import java.util.Map;
-
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
-import org.osgi.service.component.annotations.ReferenceCardinality;
-import org.osgi.service.component.annotations.ReferencePolicy;
-import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.osgi.service.component.annotations.Deactivate;
 
 /**
  * @author Michael C. Han
  */
 @Component(service = NodeExecutorRegistry.class)
 public class NodeExecutorRegistry {
 
-	public NodeExecutor getNodeExecutor(String nodeTypeString)
-		throws KaleoDefinitionValidationException {
-
-		return _nodeExecutors.getNodeTypeDependentObjects(nodeTypeString);
+	public NodeExecutor getNodeExecutor(String nodeTypeString) {
```

Comment:
@W2jx Please modify the parameter type from String to NodeType, then you do not need to do valueOf

Reply:
Hi @tinatian , In classes using method `getNodeExecutor`, the parameter passed to it is `KaleoNode.getType`. We may have to use method `NodeType.valueOf` once. Do you want me to use it before passing parameters?

Reply:
@W2jx My bad, I thought the invocations used the enum type